src/nvme/linux.c

```diff
@@ -470,9 +475,20 @@ int nvme_gen_dhchap_key(char *hostnqn, enum nvme_hmac_alg hmac,
 	memcpy(key, secret, key_len);
 	return 0;
 }
+
+long nvme_insert_tls_key(const char *keyring, const char *hostnqn,
+			 const char *subsysnqn, int hmac,
+			 unsigned char *configured_key, int key_len)
+{
+	nvme_msg(NULL, LOG_ERR, "TLS key operations not supported; "\
+		 "recompile with OpenSSL support.\n");
+	return -EOPNOTSUPP;
```

Comment:
```c
errno=-EOPNOTSUPP;
return 1;
```

Reply:
Just noticed that we are using -1 as error code in this part of the API and the rest of the library 1. We really should ditch the `errno/return 1` stupidity in future.

Reply:
I've now changed the function to return the key serial (of type 'long') or '0' on failure.